toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +21,75 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file_and_check(path, filename, expected_signature) -> Optional[bool]:
```

Comment:
We're overloading the meaning of the return value for `find_file_and_check` where true/false say if the check was OK and `None` indicates the file was missing. To make function have a single responsibility, what do you think about extracting the `Path(path_to_check).is_file()` check outside the function?

Reply:
this code doesn't really bother me much given how small it is.

but if needed, i could change the logic so that there was a find-file that returned an Optional[Path] and then check the signature of that path.

Reply:
separated search from signature-check